File: Rider/ThirdTask/Laba8sharp/CodingNewGenerations/Model/Entity/Lights.cs
using System.Text;
using Laba8sharp.CodingNewGenerations.Model.Abstracts;

namespace Laba8sharp.CodingNewGenerations.Model.Entity
{
    public class Lights : Toy
    {
        private int lifeTime;
        private int usualTemp;
        private int colorfulTemperature;
        
        public Lights()
        {
        }

        public Lights(double price, double weight, double size, int lifeTime, int usualTemp, int colorfulTemperature) : base(price, weight, size)
        {
            this.lifeTime = lifeTime;
            this.usualTemp = usualTemp;
            this.colorfulTemperature = colorfulTemperature;
        }
        
        public int GetLifeTime() {
            return lifeTime;
        }

        public void SetLifeTime(int lifeTime) {
            this.lifeTime = lifeTime;
        }

        public int GetUsualTemp() {
            return usualTemp;
        }

        public void SetUsualTemp(int usualTemp) {
            this.usualTemp = usualTemp;
        }

        public int GetColorfulTemperature() {
            return colorfulTemperature;
        }

        public void SetColorfulTemperature(int colorfulTemperature) {
            this.colorfulTemperature = colorfulTemperature;
        }

        public override string ToString()
        {
            return new StringBuilder().Append("\nLights "+ base.ToString() +
                                              "\n\tLife time: " + lifeTime +
                                              "\n\tUsual temp: " + usualTemp +
                                              "\n\tColorful temperature: " + colorfulTemperature).ToString();
        }
        
    }
}

File: Ruby/Object-oriented-programming/Lab/Model/Entity/Lights.rb
require_relative "../../../Lab/Model/Entity/Abstracts/Toy"

class Lights < Toy


  def initialize(price, weight, size, life_time, usual_temp, colorful_temperature)
    super(price, weight, size)
    @life_time, @usual_temp, @colorful_temperature = life_time, usual_temp, colorful_temperature
  end

  attr_accessor :life_time, :usual_temp, :colorful_temperature

  def to_s
    Lights.name + super + "\nLife time = " + @life_time.to_s + "\nUsual temp = " + @usual_temp.to_s + "\nColorful temperature = " + @colorful_temperature.to_s
  end

end
